Loader/SymfonyFixturesLoader.php

```diff
@@ -72,6 +84,48 @@ protected function createFixture($class)
         return $this->loadedFixtures[$class];
     }
 
+    /**
+     * Returns the array of data fixtures to execute.
+     *
+     * @param string[] $groups
+     *
+     * @return FixtureInterface[]
+     */
+    public function getFixtures(array $groups = [])
+    {
+        $fixtures = parent::getFixtures();
+
+        if (empty($groups)) {
+            return $fixtures;
+        }
+
+        $filteredFixtures = [];
+        foreach ($fixtures as $key => $fixture) {
+            foreach ($groups as $group) {
+                if (isset($this->groupsFixtureMapping[$group][get_class($fixture)])) {
+                    $filteredFixtures[$key] = $fixture;
```

Comment:
`yield` ?

Reply:
instead of building and returning an array, as it's gonna be foreach'd, using a generator would make more sense imo.

`yield $key => $fixture`, and replace the early return with a `yield from $fixtures;` and `return;`.

So this method would return `void`. :}

Reply:
Then you would also have to change the same method in the parent class `Doctrine\Common\DataFixtures\Loader`

That would be a big change across different repositories. It would also be a BC break.
But if you want to advacte for this change, I guess your best option would be to create a separate issue about it.

Reply:
Oh, didn't know it was in the parent. Carry on then. :}

Reply:
That would be a BC break, which would have to come from doctrine/fixtures itself. Feel free to suggest this there 👍 
Also, the method wouldn't return `void` but rather `\Generator` - you can't use a `void` return value.